blockchain/chain_sync/src/sync.rs

```diff
@@ -168,27 +170,40 @@ where
                     }
                 }
                 NetworkEvent::PeerDialed { peer_id } => {
-                    let heaviest = self.chain_store.heaviest_tipset().unwrap();
-                    self.network
+                    let heaviest = chain_syncer.chain_store.heaviest_tipset().await.unwrap();
+                    chain_syncer.network
                         .hello_request(
                             peer_id,
                             HelloMessage {
                                 heaviest_tip_set: heaviest.cids().to_vec(),
                                 heaviest_tipset_height: heaviest.epoch(),
                                 heaviest_tipset_weight: heaviest.weight().clone(),
-                                genesis_hash: self.genesis.blocks()[0].cid().clone(),
+                                genesis_hash: chain_syncer.genesis.blocks()[0].cid().clone(),
                             },
                         )
                         .await
                 }
+                NetworkEvent::PubsubMessage { source, topics, message } => {
+                    match topics[0].as_str() {
+                       "/fil/blocks/interop"  => {
```

Comment:
the topic depends on the network name from genesis (in case that helps debugging)

Reply:
Oh, yeah. I will polish this more later. Just wanted to make it clear that we call sync when we receive something over gossip.